providers/oidc.go

```diff
@@ -279,7 +294,24 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 type OIDCClaims struct {
 	rawClaims         map[string]interface{}
 	UserID            string
-	Subject           string `json:"sub"`
-	Verified          *bool  `json:"email_verified"`
-	PreferredUsername string `json:"preferred_username"`
+	Subject           string   `json:"sub"`
+	Verified          *bool    `json:"email_verified"`
+	PreferredUsername string   `json:"preferred_username"`
+	Groups            []string `json:"groups"`
```

Comment:
I think since you are dynamically getting the groups based on the `p.GroupsClaim` above in the `rawClaims` and setting it directly, we don't want to Unmarshal it here with the JSON struct tag.

So I'm thinking you ditch the ``json:"groups"`` and have it look similar to `UserID` above (which really should be called `Email` IMO 😄 )

Reply:
Where do I unmarshal that one then? within `containsGroups`?

Reply:
Aren't you already handling it with this rawClaims logic here: https://github.com/oauth2-proxy/oauth2-proxy/pull/616/files#diff-0e18159d630972da194b7289c16d1b99R260

Reply:
Looking at UserID that does some more stuff by unmarshaling into `respJSON`, I could do something similar but do we get any benefits?

Reply:
I'm with @NickMeves, since you're extracting it using the raw calims above, you don't need to unmarshal it here. I think if you do, you limit the claim name to always being `groups` and the `--oidc-groups-claim` flag becomes redundant

Reply:
This should be done in my latest push (coming soon).

Reply:
Was this resolved? Can we add a test case to make sure that if I have two sets of strings in token, one that is `groups` and one that is `extra_groups` or something, and the groups claim is `extra_groups`, that we get the right values, I'm still concerned that this will extract the wrong groups, especially if `groups` is present but the groups claim isn't present